modules/nw-running-dpdk-rootless-tap.adoc

```diff
@@ -0,0 +1,260 @@
+
+When applying the configuration specified in a `SriovNetworkNodePolicy` object, the SR-IOV Operator may drain the nodes, and in some cases, reboot nodes.
+It may take several minutes for a configuration change to apply.
+Ensure that there are enough available nodes in your cluster to handle the evicted workload beforehand.
+
+After the configuration update is applied, all the pods in `openshift-sriov-network-operator` namespace will change to a `Running` status.
+=====
+
+. Create the `SriovNetworkNodePolicy` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f intel-dpdk-node-policy.yaml
+----
+
+. Create the following `SriovNetwork` object, and then save the YAML in the `intel-dpdk-network.yaml` file.
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetwork
+metadata:
+  name: intel-dpdk-network
+  namespace: openshift-sriov-network-operator
+spec:
+  networkNamespace: <target_namespace>
+  ipam: |-
+# ... <1>
+  vlan: <vlan>
+  resourceName: intelnics
+----
+<1> Specify a configuration object for the IPAM CNI plugin as a YAML block scalar. The plugin manages IP address assignment for the attachment definition.
++
+[NOTE]
+=====
+See the "Configuring SR-IOV additional network" section for a detailed explanation on each option in `SriovNetwork`.
+=====
++
+An optional library, app-netutil, provides several API methods for gathering network information about a container's parent pod.
+
+. Create the `SriovNetwork` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f intel-dpdk-network.yaml
+----
+
+. Create a YAML file `tap-example.yaml` defining a network attachment definition, such as in the following example:
++
+[source,yaml]
+----
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+ name: tap-one
+ namespace: <target_namespace> <1>
+spec:
+ config: '{
+   "cniVersion": "0.4.0",
+   "name": "tap",
+   "plugins": [
+     {
+        "type": "tap",
+        "multiQueue": true,
+        "selinuxcontext": "system_u:system_r:container_t:s0",
+        "owner": 1001,
+        "group": 1001
+     },
+     {
+       "type":"tuning",
+       "capabilities":{
+         "mac":true
+       }
+     }
+   ]
+ }'
+----
+<1> Specify the same `target_namespace` where the `SriovNetwork` object `intel-dpdk-network` is created.
+
+. Apply the `yaml` by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f tap-example.yaml
+----
++
+.Example output
+[source,terminal]
+----
+networkattachmentdefinition.k8s.cni.cncf.io/tap created
+----
+
+. Create the following `Pod` spec, and then save the YAML in the `intel-dpdk-pod-rootless.yaml` file.
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dpdk-app
+  namespace: <target_namespace> <1>
+  annotations:
+    k8s.v1.cni.cncf.io/networks: intel-dpdk-network
+spec:
+  containers:
+  - name: testpmd
+    image: <DPDK_image> <2>
+    securityContext:
```

Comment:
You need to configure also podSecurityContext. Please take a look on working pod manifest example. This will work for mlx only:

```
apiVersion: v1
kind: Pod
metadata:
  name: dpdk-app
  namespace: test-namespace
  annotations:
    k8s.v1.cni.cncf.io/networks: '[
      {"name": "sriov-network", "namespace": "test-namespace"},
      {"name": "tap-one", "interface": "ext0", "namespace": "test-namespace"}]'
spec:
  nodeSelector:
    kubernetes.io/hostname: "worker-1"
  securityContext:
    fsGroup: 801
    runAsGroup: 1001
    seccompProfile:
      type: RuntimeDefault
  containers:
  - name: testpmd
    image: quay.io/ocp-edge-qe/eco-gotests-rootless-dpdk:v4.14.1
    securityContext:
      capabilities:
        drop: ["ALL"]
        add:
          - IPC_LOCK
          - SYS_RESOURCE
          - NET_RAW #for mlx only
          - NET_ADMIN
      runAsUser: 1001 
      privileged: false 
      allowPrivilegeEscalation: true 
      runAsNonRoot: true 
    volumeMounts:
    - mountPath: /dev/hugepages 
      name: hugepage
    resources:
      limits:
        openshift.io/mlxnics: "1" 
        memory: "1Gi"
        cpu: "4" 
        hugepages-1Gi: "4Gi" 
      requests:
        openshift.io/mlxnics: "1"
        memory: "1Gi"
        cpu: "4"
        hugepages-1Gi: "4Gi"
    command: ["sleep", "infinity"]
  volumes:
  - name: hugepage
    emptyDir:
      medium: HugePages
```